projects/plugins/boost/app/features/image-guide/src/Images.ts

```diff
@@ -0,0 +1,148 @@
+type Image = {
+	type: 'img' | 'srcset' | 'background';
+	url: string;
+	node: Element;
+	fileSize: {
+		width: number;
+		height: number;
+		weight: number;
+	};
+};
+export async function load( domElements: Element[] ) {
+	const parsedNodes = domElements.map( async ( el: Element ) => {
+		// Handle <img> tags first.
+		if ( el.tagName === 'IMG' ) {
+			return getImg( el as HTMLImageElement );
+		}
+
+		// Check for background images
+		// in all other elements.
+		return getBackgroundImage( el );
+	}, [] );
+
+	return ( await Promise.all( parsedNodes ) ).filter( Boolean ) as Image[];
+}
+
+async function getImageSize( url ) {
+	const response = await fetch( url, { method: 'HEAD', mode: 'no-cors' } );
+	if ( ! response.url ) {
+		console.log( `Can't get image size for ${ url } likely due to a CORS error.` );
+		return -1;
+	}
```

Comment:
It might be a good idea for the PHP-side to fetch images, so that cors isn't an issue. Future PR though.

Reply:
Yep.

Reply:
`//TODO: later`